File: 645 set-mismatch/js/main.v2.js
/**
 * Leetcode 最优解
 */

/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findErrorNums = function(nums) {
  var res = [];
  for (var i = 0; i < nums.length; i++) {
    if (nums[Math.abs(nums[i]) - 1] < 0)
      res[0] = Math.abs(nums[i]);
    else
      nums[Math.abs(nums[i]) - 1] *= -1;
  }

  for (var i = 0; i < nums.length; i++) {
    if (nums[i] > 0)
      res[1] = i + 1;
  }

  return res;
}

File: 26 remove-duplicates-from-sorted-array/c/main.v2.c
/**
 * Leetcode最优解
 */
int removeDuplicates(int* nums, int numsSize) {
  if(numsSize <= 0)
      return 0;

  int j = 1;
  for(int i = 1; i < numsSize; ++i)
      if(nums[i] != nums[j - 1])
        nums[j++] = nums[i];

  return j;
}
